### ISSUE TITLE
NSObject extended with EVReflectable causes a crash

### ISSUE THREAD

**Issue by Alarson93** [evermeer/EVReflection#245]
I have a EVReflactable realm Object, called PersonRealm, that is based off of this response. The deserialization is currently working perfectly! But, I'm also working on a MyError object (also EVReflectable) that conforms to a schema. A very basic example:
`class MyError: NSObject, EVReflectable {
    var code: String?
}`

While working on this I accidentally tried creating a MyError object using the JSON meant for PersonRealm. I noticed that this caused a crash. If I switch MyError to a Realm Object instead of an NSObject, the problem goes away.

I can see the exception is being thrown from EVReflection.swift on line 979:
`anyObject.setValue(setValue, forKey: key)`

This line is hit regardless of whether MyError is an Object or NSObject, but strangely it only causes a crash for NSObject. Also, I realize that initializing an object with the wrong JSON isn't ideal in the first place - it just happened to reveal a potential bug.

**Update:** Apparently the issue doesn't have to do with the JSON at all. Even if I use the correct JSON, having an object that is NSObject and EVReflectable causes a crash. EVObject works just fine though (which is probably the better solution anyways). 

In short, this is bad:
`class MyError: NSObject, EVReflectable`

This is good:
`class MyError: EVObject`

**Comment by evermeer** [comment]
If you use `class MyError: EVObject` and look at the console output, then you will see an error reported by EVReflection. That should give you an indication what's going wrong when you only use `anyObject.setValue(setValue, forKey: key)` The difference between these 2 is that the EVObject has a couple of functions implemented that you did not do yourself. It's probably the function `setValue forUndefinedKey`. The main reason for errors that end up there are that you are using an unsupported type like a nullable int. See the readme to see how you can workaround those. https://github.com/evermeer/EVReflection#known-issues

**Comment by Alarson93** [comment]
I chose a reasonable JSON body for MyError in case something about the Person JSON was causing an issue (like a nil int). For MyError I attempted to pass in this JSON:
`{
	"code": "Oh no!"
}`

If I change the type to EVObject, it succeeds. There is no message in the console. If I attempt to deserialize this JSON as NSObject, EVReflectable the app crashes. 

On a side note - your comment about nullable ints reminded me to create a converter for RealmOptionals. Here are my converters if you are interested:
![screen shot 2017-10-17 at 12 14 21 pm](https://user-images.githubusercontent.com/16312918/31678773-c5fe50c4-b334-11e7-8677-db4f1bfb224c.png)
(Pasting the code from the screenshot breaks the GitHub markdown for some reason.)

**Comment by Alarson93** [comment]
I see that Realm 3 has been released. We no longer need to wrap primitives into Realm objects (like RealmString).

Are you likely to add support for deserializing primitive arrays to Lists? Right now I have to explicitly write out a transformer/convert, which made sense for when we had to wrap primitives, but we could probably skip that step entirely now.

To clarify, it would be awesome to be able to populate this object without defining property converters:
 @`
class MyObject: Object, EVReflectable {
    let urls = List<String>()
    let my_int = RealmOptional<Int>()
}
`

That support would effectively make my above code (from the screenshot) irrelevant and greatly improve the ease of use (when dealing with Realm).

**Comment by evermeer** [comment]
I will have a look at it this weekend. This week I already made EVReflection Realm 3 compatible. See the latest version.

**Comment by Alarson93** [comment]
Yup! I just updated yesterday :)

I look forward to hearing about how it goes! (looking at it this weekend that is)

**Comment by evermeer** [comment]
I just pushed version 5.1.4 with support for primitive List in Realm. I will now have a look at the optionals.

**Comment by evermeer** [comment]
And I just pushed version 5.1.5 with support for RealmOptional.

For more info see
https://github.com/evermeer/EVReflection/tree/master/Source/Realm
Or see the Realm unit tests.

If you have any other issues just let me know.

**Comment by Alarson93** [comment]
I just tested it and it worked beautifully! Thank you very much.
